CMakeLists.txt

```diff
@@ -283,36 +309,48 @@ if(SKBUILD) # Terra Addon build
 	add_subdirectory(src/open_pulse)
 else() # Standalone build
 
-	# We build SIMD filed separately, because they will be reached only if the
-	# machine running the code has SIMD support
 	if(CMAKE_HOST_SYSTEM_PROCESSOR STREQUAL "x86_64" OR CMAKE_HOST_SYSTEM_PROCESSOR STREQUAL "AMD64")
-		set(SIMD_SOURCE_FILE "${PROJECT_SOURCE_DIR}/src/simulators/statevector/qv_avx2.cpp")
-		if(APPLE OR UNIX)
-			set_source_files_properties(${SIMD_SOURCE_FILE} PROPERTIES COMPILE_FLAGS "-mfma -mavx2")
-		elseif(MSVC)
-			set_source_files_properties(${SIMD_SOURCE_FILE} PROPERTIES COMPILE_FLAGS "/arch:AVX2")
-		endif()
-	endif()
-
-	set(AER_SIMULATOR_SOURCES "${PROJECT_SOURCE_DIR}/contrib/standalone/qasm_simulator.cpp"
-		"${SIMD_SOURCE_FILE}")
-	set_source_files_properties(${AER_SIMULATOR_SOURCES} PROPERTIES CUDA_SOURCE_PROPERTY_FORMAT OBJ)
+	  # We build SIMD filed separately, because they will be reached only if the
+	  # machine running the code has SIMD support
+	  set(SIMD_SOURCE_FILE "${PROJECT_SOURCE_DIR}/src/simulators/statevector/qv_avx2.cpp")
+  endif()
 
+	set(AER_SIMULATOR_SOURCES "${PROJECT_SOURCE_DIR}/contrib/standalone/qasm_simulator.cpp")
 	set(CMAKE_INSTALL_RPATH_USE_LINK_PATH TRUE)
-	if(CUDA_FOUND)
-		cuda_add_executable(qasm_simulator ${AER_SIMULATOR_SOURCES})
+	if(CUDA_FOUND AND AER_THRUST_BACKEND STREQUAL "CUDA")
+		set_source_files_properties(${SIMD_SOURCE_FILE} PROPERTIES LANGUAGE CUDA)
+		set_source_files_properties(${AER_SIMULATOR_SOURCES} PROPERTIES LANGUAGE CUDA)
+		set_source_files_properties(${AER_SIMULATOR_SOURCES} PROPERTIES COMPILE_FLAGS "${CUDA_NVCC_FLAGS}")
+                set(L ${SIMD_FLAGS_LIST})
+  		list(TRANSFORM L PREPEND " --compiler-options ")
+  		string(REPLACE ";" " " SIMD_FLAGS ${L})
```

Comment:
It would be great for readability and maintainability to wrap in a function these kind of transformations that are used in several places and in different CMakeFiles